### ISSUE TITLE
Feature Request: Allow access to (unparsed) annotations

### ISSUE THREAD

**Issue by billdenney** [metrumresearchgroup/mrgsolve#594]
I often name my annotations like "`log(Intercompartmental clearance central-peripheral 1 (compartment volume/time))`".  And, I would like to use this naming to link my original model in NONMEM to the mrgsolve model for purposes of parameter linkage (I don't like using `THETA*` names as they are not descriptive and are therefore error prone).

For that, is there a preferred way to extract annotation values from the model?  The only way that I see to do so is by accessing the slots directly which is usually fragile, like `model@annot`.

Additionally, given the way that I write my annotations (and also related to #578), sometimes they do not parse as expected.  Could the raw annotation also be stored so that I can match up between the original model and the mrgsolve model?  With the example above, it parses to the following structure which I understand why it looks the way it does, but it's not feasible to directly match to the input.  Maybe a new `raw` or `original` column that contains the unparsed input would help here.

```
> model_raw@annot$data[7,]
  block name descr                                                                     unit options
7 PARAM  lQ1  log) Intercompartmental clearance central-peripheral 1compartment volume/time       .
```

**Comment by kylebaron** [comment]
That parsed annotation was bound to fail / break / disappoint.  Obviously we can't just do away with that whole system at this point.  But I started on re-imagining this more-structured input method, where it might mean more typing on the part of the user, but there won't be any parsing etc. 

This is WIP to say the least, but something along the lines of:

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(mrgsolve)
#> 
#> Attaching package: 'mrgsolve'
#> The following object is masked from 'package:stats':
#> 
#>     filter

code <- '
$yaml 
param: 
  cl: {value: 1.23, descr: log(clearance), unit: L/h}
  vc: [4.56, volume, L] 
cmt:
  gut: [default dosing compartment, nmol]
  cent: [central compartment, nmol]
init:
  foo: [1, some other compartment]
'

mod <- mcode("foo", code)
#> Building foo ...
#> done.

mod
#> 
#> 
#> -----------------  source: foo.cpp  -----------------
#> 
#>   project: /private/var/fol.../T/RtmpMXailQ
#>   shared object: foo-so-13025326924d6 
#> 
#>   time:          start: 0 end: 24 delta: 1
#>                  add: <none>
#> 
#>   compartments:  gut cent foo [3]
#>   parameters:    cl vc [2]
#>   captures:       <none>
#>   omega:         0x0 
#>   sigma:         0x0 
#> 
#>   solver:        atol: 1e-08 rtol: 1e-08 maxsteps: 20k
#> ------------------------------------------------------

mod %>% (mrgsolve:::details)
#> # A tibble: 5 x 6
#>   block name  descr                      value.x unit  value.y
#>   <chr> <chr> <chr>                        <dbl> <chr>   <dbl>
#> 1 PARAM cl    log(clearance)                1.23 L/h      1.23
#> 2 PARAM vc    volume                        4.56 L        4.56
#> 3 CMT   gut   default dosing compartment    0    nmol     0   
#> 4 CMT   cent  central compartment           0    nmol     0   
#> 5 INIT  foo   some other compartment        1    .        1
```

<sup>Created on 2019-11-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>


You can see that something funky is happening with the annotation table.  But should be easily fixed.

**Comment by billdenney** [comment]
I’ve been thinking that moving to a structured annotation with something like json comments is the way that we should go...

That would be a massive headache, in general, and there would have to be additional prechecks of a model so that the json would be confirmed correct, and overall, it probably isn’t a good idea.

**Comment by billdenney** [comment]
Hey Kyle, I encountered this again today with the attached example.  Since I know what the issue is, I know how to fix it, but I thought that a light ping may help...

Rename this to .Rmd: [test.txt](https://github.com/metrumresearchgroup/mrgsolve/files/4784136/test.txt)

``` r
library(mrgsolve)
#> 
#> Attaching package: 'mrgsolve'
#> The following object is masked from 'package:stats':
#> 
#>     filter
mrg_model <- mread_cache("test.Rmd", soloc=".mrgsolve/")
#> Error: improper annotation format
#>  input: KA               : 0.02062938 : Typical monoclonal absorption rate From doi: 10.1208/s12248-012-9367-0 (absorption half-life of 1.4 days), Ka
#>  context: parse annotated parameter block (1)
```

<sup>Created on 2020-06-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by kylebaron** [comment]
@billdenney this is fixed (with #728 ). Please test to make sure it suites your needs. If not, please comment and I'll re-open.

**Comment by billdenney** [comment]
Thanks!
